crates/syntax/src/ast/token_ext.rs

```diff
@@ -18,12 +18,33 @@ impl ast::Comment {
     }
 
     pub fn prefix(&self) -> &'static str {
-        let &(prefix, _kind) = CommentKind::BY_PREFIX
-            .iter()
-            .find(|&(prefix, kind)| self.kind() == *kind && self.text().starts_with(prefix))
-            .unwrap();
+        let &(prefix, _kind) = CommentKind::with_prefix_from_text(self.text());
         prefix
     }
+
+    pub fn kind_and_prefix(&self) -> &(&'static str, CommentKind) {
```

Comment:
I mainly added that function as I thought it would be wasteful to iterate twice over the slice to get the two things separately.

Reply:
It's much easier to oprimize the impl after the fact than to optimize the API. Once we'll get const-fn powerful enough, this all becomes guaranteed no-op at runtime (I expect it might already be a no-op with optimizations)